[NGUYEN MINH DUC] iP

k1b1t0 opened the pull request.

k1b1t0 committed:
Level-0

k1b1t0 committed:
Level-0

k1b1t0 committed:
Add echo for chatbot

k1b1t0 committed:
Add to do list and function add to list

k1b1t0 committed:
mark/unmark tasks

k1b1t0 committed:
Ensure Java Standards

k1b1t0 committed:
Add event, todo, task

k1b1t0 committed:
Add semi-auto test

k1b1t0 committed:
refactor SuperIdol class for better readability

gbinw128 commented on src/main/java/SuperIdol.java:
```diff
@@ -0,0 +1,156 @@
+                addEvent(command);
+                break;
+            default:
+                addTask(command);
```
Nice, clearly named functions

gbinw128 commented on src/main/java/SuperIdol.java:
```diff
@@ -0,0 +1,156 @@
+            Event event = new Event(task, startTime, endTime);
+            addToList(event, task);
+        }
+        else {
```
"else" should be right after the bracket in the same line, applies to all "else"

roydenlyr replied:
else should be beside close bracket of if block

roydenlyr commented on src/main/java/SuperIdol.java:
```diff
@@ -0,0 +1,156 @@
+                break;
+            case "event":
+                addEvent(command);
+                break;
```
suggest using enums for cases

Amanda-HUANG88 commented:
Nice design ! One reminder is that else term should be immediately follow the `if` statement‘s braces in SuperIdol.java.

gbinw128 commented on src/main/java/SuperIdol.java:
```diff
@@ -0,0 +1,156 @@
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
```
Nice use of other lib utils

roydenlyr commented on src/main/java/SuperIdol.java:
```diff
@@ -0,0 +1,156 @@
+    }
+
+    public static void addEvent(String command) {
+        Pattern pattern = Pattern.compile("(event)\\s(.+?)\\s/from(.+?)\\s/to(.+)");
```
This looks like generated code

gbinw128 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+        this.task = task;
+    }
+
+    public void mark() {
+        this.isDone = true;
+    }
+
+    public void unmark() {
+        this.isDone = false;
+    }
```
Clearly named functions

gbinw128 reviewed (commented):
Overall, nicely done

roydenlyr commented on src/main/java/SuperIdol.java:
```diff
@@ -0,0 +1,156 @@
+                addDeadline(command);
+                break;
+            case "event":
+                addEvent(command);
```
Good naming convention. Each branch is short and clear

roydenlyr reviewed (commented):
Overall good coding standard. Good Job!

k1b1t0 committed:
Fix bug: add infinity task

k1b1t0 committed:
Add error handling

todo, deadline, event without task
mark, unmark without index
add some instructions for user

k1b1t0 committed:
refactor code to packages

k1b1t0 committed:
Merge branch 'branch-Level-5'

that is the requirement

k1b1t0 committed:
Merge all branch Level-5

k1b1t0 committed:
Merge all branch Level-5

k1b1t0 committed:
Add delete feature

adoorknob commented on src/main/java/superidol/task/Deadline.java:
```diff
@@ -0,0 +1,15 @@
+package superidol.task;
+
+public class Deadline extends Task{
+    private String time;
```
the deadline is a date, so this variable naming might be confusing

adoorknob commented on src/main/java/superidol/ui/SuperIdol.java:
```diff
@@ -0,0 +1,178 @@
+            try {
+                String commandKeyWord = command.split(" ")[0];
+                switch (commandKeyWord) {
+                case "exit":
```
you could have these command words as consts or enums

adoorknob commented on src/main/java/superidol/ui/SuperIdol.java:
```diff
@@ -0,0 +1,178 @@
+        }
+    }
+
+    public static void reponse() {
```
name of method should be a verb! could be clearer as well, this seems to be a response to invalid input

adoorknob commented on src/main/java/superidol/ui/SuperIdol.java:
```diff
@@ -0,0 +1,178 @@
+            + "try: event <task> /from <start> /to <end>");
+    }
+
+    public static void greeting() {
```
i like that you've separated the printing out into different methods! you could go one step further by making a separate class for this as well

adoorknob commented on src/main/java/superidol/ui/SuperIdol.java:
```diff
@@ -0,0 +1,178 @@
+            Deadline newTask = new Deadline(deadline, time);
+            addToList(newTask, deadline);
+        } else {
+            talk("Wrong input! Try \"deadline <task> /by <time>\"");
```
you can avoid using magic strings and define them at the start of the file instead

adoorknob commented on src/main/java/superidol/ui/SuperIdol.java:
```diff
@@ -0,0 +1,178 @@
+
+public class SuperIdol {
+
+    private static Task[] toDoList = new Task[100];
```
should be named plural (also easy to confuse with the `Task` object

adoorknob commented on src/main/java/superidol/ui/SuperIdol.java:
```diff
@@ -0,0 +1,178 @@
+            String command =  in.nextLine().trim();
+            try {
+                String commandKeyWord = command.split(" ")[0];
+                switch (commandKeyWord) {
```
this is very readable!

adoorknob reviewed (commented):
🎊 could try further separation into classes and avoid using magic literals, but otherwise good job! LGTM

k1b1t0 committed:
Add saving feature

k1b1t0 committed:
Create tag file

k1b1t0 committed:
Refactor code

k1b1t0 committed:
Refactor command pattern

k1b1t0 committed:
Add time feature

k1b1t0 committed:
Level-9

k1b1t0 committed:
Add find feature

k1b1t0 committed:
Add JavaDoc

k1b1t0 committed:
Merge branch 'master' of https://github.com/k1b1t0/ip

# Conflicts:
#	src/main/java/superidol/task/Task.java
#	src/main/java/superidol/tasklist/TaskList.java
#	src/main/java/superidol/ui/Ui.java

k1b1t0 committed:
Modify README.md

k1b1t0 committed:
Update README.md

k1b1t0 committed:
Update README.md

k1b1t0 committed:
Fix typo

k1b1t0 committed:
Merge pull request #1 from k1b1t0/branch-Clean-Code

Fix typo
